cpp/src/dictionary/set_keys.cu

```diff
@@ -85,13 +85,10 @@ struct dispatch_compute_indices {
     return result;
   }
 
-  template <typename Element>
+  template <typename Element, typename... Args>
   typename std::enable_if_t<!cudf::is_relationally_comparable<Element, Element>(),
                             std::unique_ptr<column>>
-  operator()(dictionary_column_view const& input,
-             column_view const& new_keys,
-             rmm::cuda_stream_view stream,
-             rmm::mr::device_memory_resource* mr)
+  operator()(Args&&...)
   {
     CUDF_FAIL("list_view dictionary set_keys not supported yet");
```

Comment:
is this the right error message? (SFINAE condition will satisfy more types than list_view)

Reply:
It is a good question. I didn't relax the SFINAE at all so I presumed the original logic that only `list_view` fails the `is_relationally_comparable<Element, Element>()` check.

If that is not the case I can modify the error message

Reply:
Looking at `traits_test.cpp` I am not seeing any special cases outside `list_view`, but maybe it wasn't updated for other `view` types.

Reply:
So `is_relationally_comparable` handles more than just `list_view` so the error message should just say something like
`dictionary set_keys not supported for this column type`.